zookeeper-api/src/test/java/org/apache/helix/zookeeper/impl/client/TestRawZkClient.java

```diff
@@ -868,59 +864,38 @@ public void testAsyncWriteOperations() {
    * Tests getChildren() when there are an excessive number of children and connection loss happens,
    * the operation should terminate and exit retry loop.
    */
-  @Test
+  @Test(timeOut = 30L)
```

Comment:
why do we need this? is it 30sec?

Reply:
`30 * 1000L ms`.

A good practice would be: if the test relies on an external system(zk here), it'd be good to have a timeout. Otherwise, if the connection hangs, the test also hangs. Eg., if the operation `multi()` hangs since it is sync, it hangs there forever?

Timeout makes the test process continue for other tests.

Reply:
Another thing is that in these 4 secs, is it possible another test would time out the session of _zkClient. Shall we use a dedicated ZkClient by creating one and later close it.

Reply:
also for safety, make it 60secs?

Reply:
This _zkClient is created within this class. While this test is running, there is no other thread referencing this _zkClient to close it.

I considered using another zkClient. But it just adds one more connection to so for this case there will be two connections at the same time. I believe reuse this one is good enough.

I ran it in my laptop as well, 30s is good enough. If it's more than 30s, something is wrong and it should timeout to fail.

Reply:
You laptop runs on SSD. Desktop or TMC machines can be slower. Leave some room for error is safe here IMO.

Reply:
30s has already good room for this test. In my tests, it completes in ~5s. If it really exceeds 30s, maybe 60s don't even help.

Reply:
Can you the whole test under helix in desktop to make sure this 30s is safe?

Reply:
I did zookeeper-api in both desktop and laptop. Is this what you meant?

Reply:
No, run all test from helix. That is what TMC is doing I guess. This is to simulate that test environment. We don't want that you fix this one but later it turned out this test still not working over there. Note the cost over there is way higher than changing it here. 

In fact, I noticed in our test, there are failed test due to Timed out from time to time. I don't think it is a good idea to add a time out here. 

By the way, can you re-open this comment thread.

Reply:
I actually didn't resolve this thread... I intended to keep it open for discussion.

Running all helix tests is not necessary. This is a separate module. Even running in the test machine, tests in different modules are independent.

Timeout is good practice. Just an example. https://github.com/apache/zookeeper/blob/fe940cdd8fb23ba09684cefb73233d570f4a20fa/zookeeper-server/src/test/java/org/apache/zookeeper/server/ZooKeeperServerMainTest.java#L165

How could you ensure 60s would not timeout then? Then how about 10 minutes, 1 hour?

Reply:
Yes, then remove this timeout. The test has a default timeout 300sec. In my opinion, relying on 300sec default is much safer. Testing under your own laptop may not fit a lot other environment.

See this:

[ERROR] testLostZkConnection(org.apache.helix.integration.TestZkConnectionLost) Time elapsed: 300.015 s <<< FAILURE!
org.testng.internal.thread.ThreadTimeoutException: Method org.testng.internal.TestNGMethod.testLostZkConnection() didn't finish within the time-out 300000

Reply:
Please let me remind you that the default timeout 300sec is set only for helix-core's tests, not in zookeeper-api.

Reply:
where is the place to config the 300secs. Shall we just configure the same in zooscalability-api?

Reply:
It is in test config file: `testng.xml`

Configuring the timeout for test suit for zookeeper-api is out of scope of this PR. If we want to do that, I would prefer to do it in another PR.